ui/app/common/displaycontrols/observation/services/formRecordTreeBuildService.js

```diff
@@ -156,12 +156,8 @@ angular.module('bahmni.common.displaycontrol.observation')
                     }
                 };
                 var translationKey = column.translationKey;
-                if (translationData && translationData.labels && translationData.labels[translationKey][0] !== translationKey) {
-                    obsGroup.concept.shortName = translationData.labels[translationKey][0];
-                }
-                else {
-                    obsGroup.concept.shortName = column.value;
-                }
+                var defaultShortName = column.value;
+                self.setTranslatedShortName(translationData, translationKey, obsGroup, defaultShortName);
```

Comment:
Hi @vinishad, wouldn't it be better to return the **short name** from **setTranslatedShortName** (though method name might change)
```suggestion
                obsGroup.concept.shortName = self.getTranslatedShortName(translationData, translationKey, defaultShortName);
```

Reply:
@sumanmaity112 thanks for the comments. Yes it would be better will update it.